From 8e03ab6a121b4f280f26412b42a532ee14df265b Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Wed, 14 Dec 2022 16:05:01 +0100
Subject: [PATCH] fsl-imx8mm-var-dart: Port pinmux for NRT board to kernel 
 5.15.60

Changes are according to "New OS ... 1 - v1.2.0.pdf"
visible in Front

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 ...imx8mm-var-dart-dt8mcustomboard-legacy.dts |  12 --
 .../imx8mm-var-dart-dt8mcustomboard.dts       |  38 +----
 .../boot/dts/freescale/imx8mm-var-dart.dtsi   | 136 ++++--------------
 3 files changed, 26 insertions(+), 160 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mm-var-dart-dt8mcustomboard-legacy.dts b/arch/arm64/boot/dts/freescale/imx8mm-var-dart-dt8mcustomboard-legacy.dts
index b1a6fe56c92a..edd6466fd809 100644
--- a/arch/arm64/boot/dts/freescale/imx8mm-var-dart-dt8mcustomboard-legacy.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mm-var-dart-dt8mcustomboard-legacy.dts
@@ -46,10 +46,6 @@ &can0 {
 	/delete-property/ microchip,rx-interrupt;
 };
 
-&ft5x06_ts {
-	reset-gpios = <&gpio4 5 GPIO_ACTIVE_LOW>;
-};
-
 &ov5640_mipi1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_csi1>;
@@ -76,14 +72,6 @@ pinctrl_can: cangrp {
 			MX8MM_IOMUXC_GPIO1_IO06_GPIO1_IO6		0x16
 		>;
 	};
-
-	pinctrl_captouch: captouchgrp {
-		fsl,pins = <
-			MX8MM_IOMUXC_SAI1_RXD3_GPIO4_IO5		0x16
-			MX8MM_IOMUXC_GPIO1_IO14_GPIO1_IO14		0x16
-		>;
-	};
-
 	pinctrl_gpio_keys: keygrp {
 		fsl,pins = <
 			MX8MM_IOMUXC_SAI1_TXD3_GPIO4_IO15		0x1c6
diff --git a/arch/arm64/boot/dts/freescale/imx8mm-var-dart-dt8mcustomboard.dts b/arch/arm64/boot/dts/freescale/imx8mm-var-dart-dt8mcustomboard.dts
index 76472e922bb9..9bfce231d0d9 100644
--- a/arch/arm64/boot/dts/freescale/imx8mm-var-dart-dt8mcustomboard.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mm-var-dart-dt8mcustomboard.dts
@@ -129,23 +129,6 @@ typec1_dr_sw: endpoint {
 		};
 	};
 
-	/* Capacitive touch controller */
-	ft5x06_ts: ft5x06_ts@38 {
-		compatible = "edt,edt-ft5206";
-		reg = <0x38>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_captouch>;
-		reset-gpios = <&pca6408_2 4 GPIO_ACTIVE_LOW>;
-		interrupt-parent = <&gpio1>;
-		interrupts = <14 IRQ_TYPE_EDGE_FALLING>;
-		touchscreen-size-x = <800>;
-		touchscreen-size-y = <480>;
-		touchscreen-inverted-x;
-		touchscreen-inverted-y;
-		wakeup-source;
-		status = "okay";
-	};
-
 	ov5640_mipi1: ov5640_mipi1@3c {
 		compatible = "ovti,ov5640_mipi";
 		reg = <0x3c>;
@@ -327,13 +310,7 @@ &uart2 {
 	status = "okay";
 };
 
-/* Header */
-&uart3 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart3>;
-	status = "okay";
-};
-
+/* uart3 TXD used by gpio5 27 */
 &usbotg1 {
 	dr_mode = "otg";
 	hnp-disable;
@@ -462,13 +439,6 @@ MX8MM_IOMUXC_UART2_TXD_UART2_DCE_TX		0x140
 		>;
 	};
 
-	pinctrl_uart3: uart3grp {
-		fsl,pins = <
-			MX8MM_IOMUXC_UART3_RXD_UART3_DCE_RX		0x140
-			MX8MM_IOMUXC_UART3_TXD_UART3_DCE_TX		0x140
-		>;
-	};
-
 	pinctrl_reg_usdhc2_vmmc: regusdhc2vmmc {
 		fsl,pins = <
 			MX8MM_IOMUXC_SD2_RESET_B_GPIO2_IO19		0xc1
@@ -523,12 +493,6 @@ MX8MM_IOMUXC_GPIO1_IO01_PWM1_OUT		0x06
 		>;
 	};
 
-	pinctrl_captouch: captouchgrp {
-		fsl,pins = <
-			MX8MM_IOMUXC_GPIO1_IO14_GPIO1_IO14		0x16
-		>;
-	};
-
 	pinctrl_extcon: extcongrp {
 		fsl,pins = <
 			MX8MM_IOMUXC_GPIO1_IO10_GPIO1_IO10		0x19
diff --git a/arch/arm64/boot/dts/freescale/imx8mm-var-dart.dtsi b/arch/arm64/boot/dts/freescale/imx8mm-var-dart.dtsi
index 23dba5d939f1..551903e24b29 100644
--- a/arch/arm64/boot/dts/freescale/imx8mm-var-dart.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mm-var-dart.dtsi
@@ -50,55 +50,6 @@ reserved-memory {
 		/* cma region is provided by kernel command line as cma=<size>M */
 		/delete-node/ linux,cma;
 	};
-
-	sound-wm8904 {
-		compatible = "simple-audio-card";
-		simple-audio-card,bitclock-master = <&dailink_master>;
-		simple-audio-card,format = "i2s";
-		simple-audio-card,frame-master = <&dailink_master>;
-		simple-audio-card,name = "wm8904-audio";
-		simple-audio-card,routing =
-			"Headphone Jack", "HPOUTL",
-			"Headphone Jack", "HPOUTR",
-			"IN2L", "Line In Jack",
-			"IN2R", "Line In Jack",
-			"IN1L", "Microphone Jack",
-			"IN1R", "Microphone Jack";
-		simple-audio-card,widgets =
-			"Microphone", "Microphone Jack",
-			"Headphone", "Headphone Jack",
-			"Line", "Line In Jack";
-		simple-audio-card,mclk-fs = <256>;
-
-		dailink_master: simple-audio-card,codec {
-			sound-dai = <&wm8904>;
-		};
-
-		simple-audio-card,cpu {
-			sound-dai = <&sai3>;
-		};
-	};
-
-	sound-spdif {
-		compatible = "fsl,imx-audio-spdif";
-		model = "imx-spdif";
-		spdif-controller = <&spdif1>;
-		spdif-out;
-		spdif-in;
-		status = "disabled";
-	};
-
-	sound-micfil {
-		compatible = "fsl,imx-audio-card";
-		model = "imx-audio-micfil";
-		pri-dai-link {
-			link-name = "micfil hifi";
-			format = "i2s";
-			cpu {
-				sound-dai = <&micfil>;
-			};
-		};
-	};
 };
 
 &A53_0 {
@@ -331,49 +282,6 @@ &i2c3 {
 	scl-gpios = <&gpio5 18 GPIO_ACTIVE_HIGH>;
 	sda-gpios = <&gpio5 19 GPIO_ACTIVE_HIGH>;
 	status = "okay";
-
-	wm8904: audio-codec@1a {
-		compatible = "wlf,wm8904";
-		#sound-dai-cells = <0>;
-		reg = <0x1a>;
-		clocks = <&clk IMX8MM_CLK_SAI3_ROOT>;
-		clock-names = "mclk";
-		DBVDD-supply = <&reg_audio>;
-		DCVDD-supply = <&ldo5_reg>;
-		AVDD-supply = <&ldo5_reg>;
-		CPVDD-supply = <&ldo5_reg>;
-		MICVDD-supply = <&ldo5_reg>;
-		num-drc-cfgs = <5>;
-		drc-cfg-names = "default", "peaklimiter", "tradition", "soft", "music";
-		drc-cfg-regs =
-			/* coded default: KNEE_IP = KNEE_OP = 0, HI_COMP = LO_COMP = 1  */
-			<0x01af 0x3248 0x0000 0x0000>,
-			/* coded default: KNEE_IP = -24, KNEE_OP = -6, HI_COMP = 1/4, LO_COMP = 1 */
-			<0x04af 0x324b 0x0010 0x0408>,
-			/* coded default: KNEE_IP = -42, KNEE_OP = -3, HI_COMP = 0, LO_COMP = 1 */
-			<0x04af 0x324b 0x0028 0x0704>,
-			/* coded default: KNEE_IP = -45, KNEE_OP = -9, HI_COMP = 1/8, LO_COMP = 1 */
-			<0x04af 0x324b 0x0018 0x078c>,
-			/* coded default: KNEE_IP = -30, KNEE_OP = -10.5, HI_COMP = 1/4, LO_COMP = 1 */
-			<0x04af 0x324b 0x0010 0x050e>;
-		gpio-cfg = <
-			0x0018 /* GPIO1 => DMIC_CLK */
-			0xffff /* GPIO2 => don't touch */
-			0xffff /* GPIO3 => don't touch */
-			0xffff /* GPIO4 => don't touch */
-		>;
-		status = "okay";
-	};
-};
-
-&sai3 {
-	#sound-dai-cells = <0>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_sai3>;
-	assigned-clocks = <&clk IMX8MM_CLK_SAI3>;
-	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
-	assigned-clock-rates = <1536000>;
-	status = "okay";
 };
 
 &spdif1 {
@@ -496,10 +404,33 @@ &vpu_v4l2 {
 };
 
 &iomuxc {
+	pinctrl-0 = <&pinctrl_gpios>;
+
+	pinctrl_gpios: gpiosgrp {
+		fsl,pins = <
+			MX8MM_IOMUXC_GPIO1_IO00_GPIO1_IO0              0x140
+			MX8MM_IOMUXC_ENET_TX_CTL_GPIO1_IO22            0x140
+			MX8MM_IOMUXC_ENET_MDIO_GPIO1_IO17              0x140
+			MX8MM_IOMUXC_ENET_MDC_GPIO1_IO16               0x140
+			MX8MM_IOMUXC_GPIO1_IO11_GPIO1_IO11             0x140
+			MX8MM_IOMUXC_SAI3_TXD_GPIO5_IO1                0x140
+			MX8MM_IOMUXC_SAI3_TXC_GPIO5_IO0                0x140
+			MX8MM_IOMUXC_SPDIF_RX_GPIO5_IO4                0x140
+			MX8MM_IOMUXC_SAI3_RXFS_GPIO4_IO28              0x140
+			MX8MM_IOMUXC_NAND_DATA00_GPIO3_IO6             0x140
+			MX8MM_IOMUXC_NAND_DATA02_GPIO3_IO8             0x140
+			MX8MM_IOMUXC_SAI5_MCLK_GPIO3_IO25              0x140
+			MX8MM_IOMUXC_SAI1_RXFS_GPIO4_IO0               0x140
+			MX8MM_IOMUXC_SAI1_RXC_GPIO4_IO1                0x140
+			MX8MM_IOMUXC_SAI2_RXFS_GPIO4_IO21              0x140
+			MX8MM_IOMUXC_SAI3_RXD_GPIO4_IO30               0x140
+			MX8MM_IOMUXC_UART3_TXD_GPIO5_IO27              0x140
+			MX8MM_IOMUXC_GPIO1_IO14_GPIO1_IO14             0x140
+		>;
+	};
+
 	pinctrl_fec1: fec1grp {
 		fsl,pins = <
-			MX8MM_IOMUXC_ENET_MDC_ENET1_MDC			0x3
-			MX8MM_IOMUXC_ENET_MDIO_ENET1_MDIO		0x3
 			MX8MM_IOMUXC_ENET_TD3_ENET1_RGMII_TD3		0x1f
 			MX8MM_IOMUXC_ENET_TD2_ENET1_RGMII_TD2		0x1f
 			MX8MM_IOMUXC_ENET_TD1_ENET1_RGMII_TD1		0x1f
@@ -511,7 +442,6 @@ MX8MM_IOMUXC_ENET_RD0_ENET1_RGMII_RD0		0x91
 			MX8MM_IOMUXC_ENET_TXC_ENET1_RGMII_TXC		0x1f
 			MX8MM_IOMUXC_ENET_RXC_ENET1_RGMII_RXC		0x91
 			MX8MM_IOMUXC_ENET_RX_CTL_ENET1_RGMII_RX_CTL	0x91
-			MX8MM_IOMUXC_ENET_TX_CTL_ENET1_RGMII_TX_CTL	0x1f
 			MX8MM_IOMUXC_GPIO1_IO09_GPIO1_IO9		0x19
 		>;
 	};
@@ -582,22 +512,9 @@ MX8MM_IOMUXC_GPIO1_IO07_GPIO1_IO7		0x41
 		>;
 	};
 
-	pinctrl_sai3: sai3grp {
-		fsl,pins = <
-			MX8MM_IOMUXC_SAI3_RXFS_SAI3_RX_SYNC		0xd6
-			MX8MM_IOMUXC_SAI3_RXC_SAI3_RX_BCLK		0xd6
-			MX8MM_IOMUXC_SAI3_RXD_SAI3_RX_DATA0		0xd6
-			MX8MM_IOMUXC_SAI3_TXFS_SAI3_TX_SYNC		0xd6
-			MX8MM_IOMUXC_SAI3_TXC_SAI3_TX_BCLK		0xd6
-			MX8MM_IOMUXC_SAI3_TXD_SAI3_TX_DATA0		0xd6
-			MX8MM_IOMUXC_SAI3_MCLK_SAI3_MCLK		0xd6
-		>;
-	};
-
 	pinctrl_spdif1: spdif1grp {
 		fsl,pins = <
 			MX8MM_IOMUXC_SPDIF_TX_SPDIF1_OUT		0xd6
-			MX8MM_IOMUXC_SPDIF_RX_SPDIF1_IN			0xd6
 		>;
 	};
 
@@ -716,16 +633,13 @@ pinctrl_flexspi0: flexspi0grp {
 		fsl,pins = <
 			MX8MM_IOMUXC_NAND_ALE_QSPI_A_SCLK		0x1c2
 			MX8MM_IOMUXC_NAND_CE0_B_QSPI_A_SS0_B		0x82
-			MX8MM_IOMUXC_NAND_DATA00_QSPI_A_DATA0		0x82
 			MX8MM_IOMUXC_NAND_DATA01_QSPI_A_DATA1		0x82
-			MX8MM_IOMUXC_NAND_DATA02_QSPI_A_DATA2		0x82
 			MX8MM_IOMUXC_NAND_DATA03_QSPI_A_DATA3		0x82
 		>;
 	};
 
 	pinctrl_pdm: pdmgrp {
 		fsl,pins = <
-			MX8MM_IOMUXC_SAI5_MCLK_SAI5_MCLK		0xd6
 			MX8MM_IOMUXC_SAI5_RXC_PDM_CLK			0xd6
 			MX8MM_IOMUXC_SAI5_RXFS_SAI5_RX_SYNC		0xd6
 			MX8MM_IOMUXC_SAI5_RXD0_PDM_DATA0		0xd6
-- 
2.37.2

